src/kurobako/CMakeLists.txt

```diff
@@ -0,0 +1,41 @@
+#not sure if this is the actual version we need, by i'm using Versiopn 3.8 features.
+cmake_minimum_required(VERSION 3.8.1)
+
+project (kurobako) #kurobako is black box
+
+#Link config file. There are better ways to do this but i'm too lazy to fully master cmake.
+configure_file(
+    "${CMAKE_SOURCE_DIR}/kurobakoconfig.h.in"
+    "${CMAKE_SOURCE_DIR}/src/kurobako/kurobakoconfig.h"
+)
+
+#Not 100% working. still figuring out linking libraries into the engine
+set(CMAKE_STATIC_LINKER_FLAGS_DEBUG "/LTCG /MACHINE:X64")
+set(CMAKE_STATIC_LINKER_FLAGS_RELEASE "/LTCG /MACHINE:X64")
+set(CMAKE_STATIC_LINKER_FLAGS_PROFILE "/LTCG /MACHINE:X64")
+set(CMAKE_STATIC_LINKER_FLAGS_FINAL "/LTCG /MACHINE:X64")
+
+#includes
+include_directories("$(CMAKE_SOURCE_DIR)/src")
+
+#Add manually each time.
+set (KUROBAKOSOURCEFILES 
+kurobakoconfig.h
+enginemain.h enginemain.cpp
+graphics/teststructure.cpp
+)
+
+#auto version
+#file (GLOB_RECURSE KUROBAKOSOURCEFILES *.h *.cpp)
```

Comment:
Should glob automatically, Don't see us having automatic file generators anytime soon